src/runtime/methodbinder.cs

```diff
@@ -633,32 +666,52 @@ static Type TryComputeClrArgumentType(Type parameterType, IntPtr argument, bool
 
             return clrtype;
         }
-
-        static bool MatchesArgumentCount(int positionalArgumentCount, ParameterInfo[] parameters,
+        /// <summary>
+        /// Check whether the number of Python and .NET arguments match, and compute additional arg information.
+        /// </summary>
+        /// <param name="pynargs">Number of positional args passed from Python.</param>
+        /// <param name="parameters">Parameters of the specified .NET method.</param>
+        /// <param name="kwargDict">Keyword args passed from Python.</param>
+        /// <param name="isOperator">True if the parameters' method is an operator.</param>
+        /// <param name="paramsArray">True if the final param of the .NET method is an array (`params` keyword).</param>
+        /// <param name="defaultArgList">List of default values for arguments.</param>
+        /// <param name="kwargsMatched">Number of kwargs from Python that are also present in the .NET method.</param>
+        /// <param name="defaultsNeeded">Number of non-null defaultsArgs.</param>
+        /// <returns></returns>
+        static bool MatchesArgumentCount(int pynargs, ParameterInfo[] parameters,
             Dictionary<string, IntPtr> kwargDict,
+            bool isOperator,
             out bool paramsArray,
             out ArrayList defaultArgList,
             out int kwargsMatched,
             out int defaultsNeeded)
         {
             defaultArgList = null;
             var match = false;
-            paramsArray = parameters.Length > 0 ? Attribute.IsDefined(parameters[parameters.Length - 1], typeof(ParamArrayAttribute)) : false;
-            var kwargCount = kwargDict.Count;
+            int clrnargs = parameters.Length;
```

Comment:
Please, don't introduce shorthands.
Same for `pynargs`.

Reply:
would `numClrArgs` and `numPyArgs` respectively be okay? I agree that the names are not intuitive, I got the idea from another part of the code so I thought maybe it was a naming convention (maybe I could change that too?): https://github.com/pythonnet/pythonnet/blob/2972f49cee96eec743d99d229886245485614d7b/src/runtime/indexer.cs#L71

Reply:
Just should not rename without necessity, as it expands lines affected by change.

Reply:
E.g. you should revert the part where you rename local and replace `parameters.Length` with `clrnargs`.

Reply:
@christabella please, rename `pynargs` back too.